### PR TITLE
Update toolchain and improve verible syntax parsing

### PR DESCRIPTION
**Pull request by Andful**

### CODE REVIEW

**Comment by Andful** [comment]
Not updating `sv-parser` due to the conversation at https://github.com/vivekmalneedi/veridian/pull/176#discussion_r1122616240

> Upgrading this is a little dangerous. I pushed my own attempt into the bump_deps branch, but the gist of it is that I think the version I was using hit a sweet spot where I could pass all my tests that involved parsing invalid code (by attempting to remove lines with syntax errors and rerunning the parse). After the upgrade that no longer seemed to be the case. There is a allow_incomplete flag now, which I think you set to true but that causes it to discard large parts of the syntax tree, which for the tests is effectively the entire tree. This might be worth the price regardless if the syntax tree it returns on "real" code is good enough for the language server to still be helpful.

> The correct solution to this issue is to use a fault-tolerant IDE parser and I have new indexing code that uses treesitter to parse in the develop branch, but I haven't had time to actually rewrite the rest of veridian to accommodate it.

File: src/diagnostics.rs [hunk 1]
```diff
@@ -199,7 +199,10 @@ fn verible_syntax(
         .arg("-")
         .spawn()
         .ok()?;
-    let re = Regex::new(r"^.+:(?P<line>.*):(?P<col>.*):\s(?P<message>.*)\s.*$").ok()?;
+    let re = Regex::new(
```
Comment (vivekmalneedi) [inline src/diagnostics.rs:202]: can you make this regex pre-compiled?
https://docs.rs/regex/latest/regex/#avoid-re-compiling-regexes-especially-in-a-loop

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -71,7 +71,7 @@ jobs:
               uses: actions-rs/toolchain@v1
               with:
                   profile: minimal
-                  toolchain: 1.52.1
+                  toolchain: 1.68.0
```
Comment (vivekmalneedi) [inline .github/workflows/ci.yml:74]: Any particular reason to put the CI at rust 1.68, I think the latest is 1.71?

File: src/definition/def_types.rs [hunk 3]
```diff
@@ -408,8 +407,9 @@ impl GenericDec {
             byte_idx: 0,
             url: url.clone(),
             type_str: String::new(),
-            completion_kind: CompletionItemKind::Variable,
-            symbol_kind: SymbolKind::Unknown,
+            completion_kind: CompletionItemKind::VARIABLE,
+            // FIXME: check if this replacement is correct
+            symbol_kind: SymbolKind::NULL,
```
Comment (vivekmalneedi) [inline src/definition/def_types.rs:412]: I'm not sure why that was unknown previously, I would assume NULL or VARIABLE are fine as replacements

**Review by vivekmalneedi — CHANGES_REQUESTED** [review]
Sorry for the delay, this looks good. 
I don't think it was able to build before when I left sv-parser at the same version and upgraded everything else so I'm not sure what changed but it looks like the tests are passing so that's great.

File: src/diagnostics.rs [hunk 4]
```diff
@@ -199,7 +199,20 @@ fn verible_syntax(
...
-    let re = Regex::new(r"^.+:(?P<line>.*):(?P<col>.*):\s(?P<message>.*)\s.*$").ok()?;
+
+    static RE: std::sync::OnceLock<Result<Regex, regex::Error>> = std::sync::OnceLock::new();
+    let re = RE
+        .get_or_init(|| {
+            Regex::new(
+                r"^.+:(?P<line>\d*):(?P<startcol>\d*)(?:-(?P<endcol>\d*))?:\s(?P<message>.*)\s.*$",
+            )
+            .map_err(|e| {
+                eprint!("{e}");
```
Comment (Andful) [inline src/diagnostics.rs:210]: Maybe it is better to panic here. The regex should be compiled only once and should work unless there is something outrageously wrong.

**Comment by Andful** [comment]
No worries!
Oh yeah! test do pass with an updated version of `sv-parser`.
Tell me if you want the commits squashed.

**Comment by Andful** [comment]
Not sure why tests are failing. Will look at this tomorrow!

**Comment by Andful** [comment]
> No worries! Oh yeah! test do pass with an updated version of `sv-parser`. Tell me if you want the commits squashed.
Nvm. They don't pass with the updated version.

**Review by vivekmalneedi — APPROVED** [review]
Looks good, thanks!

**Pull request merged**